### PR TITLE
✨♻️✅✚ proof of concept local and remote test runner

### PR DESCRIPTION
**Pull request by filmaj**
- adding wdio sauce service as a dev dependency
- starting sauce tunnel in circleci to bridge circleci and sauce environments and enable localhost based tests, but on CI
- added four e2e npm run scripts (one for running locally, one for running on sauce, and each of those for helix vs. aem)
- added an npm run ci run script to encapsulate what circleci will do
- now have four different wdio configs, one for local and one for remote, and each of these for aem vs. helix
- added docs under the `test/` directory so hopefully the options are clear

### CODE REVIEW

**Comment by filmaj** [comment]
Oh and we can slowly, over time, add more tests, we don't need to be super exhaustive with them right away.

**Comment by filmaj** [comment]
I haven't, but I'll try it out and see how it differs in this PR

**Comment by filmaj** [comment]
Lol the saucelabs orb is buggy as hell. Their 'close tunnel' command doesn't work. Gonna need some workarounds in here.

**Comment by filmaj** [comment]
I still need to fix the manual tunnel shutdown shite.

Honestly I'm tempted to just go back to handling the tunnel myself for now. The saucelabs orb hasnt been updated in four months. I filed an issue but I don't think it gets a lot of attention.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 version: 2.1
+orbs:
+  sauce-connect: saucelabs/sauce-connect@volatile
```
Comment (trieloff) [inline .circleci/config.yml]: That really makes things easier.

Comment (filmaj) [reply .circleci/config.yml]: It's good in theory but if we commit to it I have a feeling we will be the ones updating the code for it: https://github.com/saucelabs-sample-test-frameworks/CircleCI-SauceLabs-ORB - no updates in four months, unanswered issues and PRs.

Comment (trieloff) [reply .circleci/config.yml]: Looks like the guy who built it is their technical director of sales engineering, which means it's likely demo code. I've reached out to him on LinkedIn, maybe we can entice them with a deeper integration/partnership if they manage to step up their quality.

Comment (trieloff) [reply .circleci/config.yml]: I'm having a call with him tomorrow 9am PST. @filmaj – do you want to join?

Comment (filmaj) [reply .circleci/config.yml]: @trieloff yes

Comment (filmaj) [reply .circleci/config.yml]: As a former employee of Sauce Labs, is this a conflict of interest?

Comment (trieloff) [reply .circleci/config.yml]: It would be one if you also were a current one.

**Pull request merged**